packages/cip30/src/Wallet/Wallet.ts

```diff
@@ -24,67 +31,66 @@ export type RequestAccess = () => Promise<boolean>;
 
 export type WalletOptions = {
   logger?: Logger;
-  persistAllowList?: boolean;
-  storage?: Storage;
+  storage?: Storage.LocalStorageArea;
 };
 
 const defaultOptions = {
   logger: dummyLogger,
   persistAllowList: false,
-  storage: window.localStorage
+  storage: storage.local
+};
+
+type WalletStorage = {
+  allowList: string[];
 };
 
 export class Wallet {
-  readonly version: SpecificationVersion;
+  readonly apiVersion: ApiVersion;
   readonly name: WalletName;
-
-  private allowList: string[];
-  private logger: Logger;
-  private readonly options: Required<WalletOptions>;
-
-  constructor(
-    properties: WalletProperties,
-    private api: WalletApi,
-    private requestAccess: RequestAccess,
-    options?: WalletOptions
-  ) {
-    this.options = { ...defaultOptions, ...options };
+  readonly icon: WalletIcon;
+
+  #allowList: string[];
+  #logger: Logger;
+  #api: WalletApi;
+  #requestAccess: RequestAccess;
+  readonly #options: Required<WalletOptions>;
+
+  constructor(properties: WalletProperties, api: WalletApi, requestAccess: RequestAccess, options?: WalletOptions) {
+    this.apiVersion = properties.apiVersion;
+    this.enable = this.enable.bind(this);
+    this.icon = properties.icon;
+    this.isEnabled = this.isEnabled.bind(this);
     this.name = properties.name;
-    this.version = properties.version;
-    this.allowList = this.options.persistAllowList ? this.getAllowList() : [];
-    this.logger = this.options.logger;
-  }
 
-  public getPublicApi(window: WindowMaybeWithCardano) {
-    return {
-      enable: this.enable.bind(this, window),
-      isEnabled: this.isEnabled.bind(this, window),
-      name: this.name,
-      version: this.version
-    };
+    this.#api = api;
+    this.#options = { ...defaultOptions, ...options };
+    this.#logger = options?.logger || this.#options.logger;
+    this.#requestAccess = requestAccess;
   }
 
-  private getAllowList(): string[] {
-    // JSON.parse(null) seems to be legit
-    return JSON.parse(this.options.storage.getItem(this.name)!) || [];
+  async #getAllowList(storageKey: string): Promise<string[]> {
+    if (!storageKey) return Promise.resolve([]);
+    try {
+      const persistedStorage: Record<string, WalletStorage> = await this.#options.storage.get(storageKey);
+      return persistedStorage[storageKey].allowList;
+    } catch {
+      return [];
+    }
   }
 
-  private allowApplication(appName: string) {
-    this.allowList.push(appName);
-
-    if (this.options.persistAllowList) {
-      const currentList = this.getAllowList();
-      // Todo: Encrypt
-      this.options.storage?.setItem(this.name, JSON.stringify([...currentList, appName]));
-      this.logger.debug(
-        {
-          allowList: this.getAllowList(),
-          module: 'Wallet',
-          walletName: this.name
-        },
-        'Allow list persisted'
-      );
-    }
+  async #allowApplication(appName: string) {
+    this.#allowList.push(appName);
```

Comment:
I think `this.#allowList` could potentially be undefined here. Since it's only used in async methods, maybe it would be better to not directly access this, but instead have an async method or a getter to initialize it if undefined and then return it (similar to code in `isEnabled`).